src/pages/main/Main.tsx

```diff
@@ -1,25 +1,33 @@
 import { AiFillFolder, AiFillTool } from 'react-icons/ai';
+import { useContext, useEffect, useState } from 'react';
 import { Button } from '@components';
 import Editor from './Editor';
+import { UserContext } from 'context/UserContextProvider';
 import { compileDocument } from './service/compilation-service';
 import styles from './Main.module.less';
 import { useNavigate } from 'react-router-dom';
-import { useState } from 'react';
-
 
 const MainPage = () => {
+  const { user, logout } = useContext(UserContext);
+
   const [documentSource, setDocumentSource] = useState<string>('');
   const [documentUrl, setDocumentUrl] = useState<string>('example.pdf');
   const [compilationError, setCompilationError] = useState<string | null>(null);
   const [compilationLogs, setCompilationLogs] = useState<string>('');
 
   const navigate = useNavigate();
 
-  const logout = () => {
-    localStorage.removeItem('user-token');
+  const onLogoutClick = () => {
+    logout();
     navigate('/login');
   };
```

Comment:
This seems to be very common pattern. Can we include `navigate('/login')` in `logout` callback? It would simplify tests as well.

Reply:
it actually worked, this simplifies a lot of stuff. Now UserProvider redirects to login page whenever user is null